server/sonar-server/src/main/java/org/sonar/server/computation/CeModule.java

```diff
@@ -37,6 +39,8 @@ protected void configureModule() {
       CeConfigurationImpl.class,
       CeLogging.class,
       CeDatabaseMBeanImpl.class,
+      SystemInfoHttpServer.class,
+      new ProcessStateProvider("Compute Engine State"),
```

Comment:
why is the name of this a constructor argument?
I think it could and should be a private constant in `ProcessStateProvider`

it's really confusing to have an object called a provider, instanced rather than added as a class to the Pico container, and which is actually not a Provider by the definition of Pico